### PR TITLE
Support OTP 21 (and 20)

### PR DESCRIPTION
**Pull request by lucafavatellawork**
~Depends on ~#27 (merged)~ #35.~

Once this PR is merged, then PR #72 (dropping OTP 20 and keeping 21) can be rebased, finalized and - eventually - reviewed and - if ok - merged.

TODO:
- [x] Review Dialyzer warnings re `ftp`/`tftp` unknown functions (moved to separate applications in [OTP 21](https://www.erlang.org/news/123)).
  - Kept. The plan is to remove these when dropping support for OTP 20.
- [x] Review warnings re `erlang:get_stacktrace/0` deprecated in [OTP 21](https://www.erlang.org/news/123).
  - Kept. The plan is to remove these when dropping support for OTP 20.
- [x] Review eunit test outcome.
- [ ] Review increased size of release (apps: debugger, wx) with OTP 22.
  - Dropped support for OTP 22. (See #74 )
- [x] Review inclusion of apps ftp, tftp in release - with both OTP 21 and 22 (minor size increase).
  - Before OTP 21, ftp and tftp used to be included in inets. So inclusion of apps ftp and tftp from OTP 21 is not different to what was included in OTP 20. These may be optimized out but in a later step.
- [x] Review logging on OTP 21 e.g. unexpected progress reports.
  - A variable had to set before the `kernel` app started, so I pass `-kernel logger_sasl_compatible true` to all invocations of `erl`. There is no "base" sys.config included in the release so this is the best I managed to do with the current build&release tools.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ dist: trusty
 language: erlang
 otp_release:
   - 20.2
+  - 21.3.8.1
```
Comment (jacobythwaites) [inline .travis.yml]: Interesting to be so precise. I imagine 21.3 would refer to 'latest' 21.3, there should be no breaking changes you'd've thought. But I defer to your experience in this!

File: make_std.mk [hunk 2]
```diff
@@ -21,7 +21,7 @@ KERNEL_MACHINE := $(shell uname -m)
 
 # Minimum and maximum OTP versions.
 OTP_MIN := 20
-OTP_MAX := 20
+OTP_MAX := 22
```
Comment (jacobythwaites) [inline make_std.mk]: Like that you keep 20 - 22

**Comment by lucafavatellawork** [comment]
This PR is still draft, and depends on #35 , that is draft too. So the next step is clearing the TODO items in #35 (i.e. re "expressions service state key").

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,8 @@
-ARG BUILDER_BASE_IMAGE=erlang:20-slim
+ARG BUILDER_BASE_IMAGE=erlang:21-slim
```
Comment (lucafavatellawork) [inline docker/Dockerfile]: Tested using https://github.com/opensparkl/sse_cli/pull/23

File: apps/sse/ct_test/sse_SUITE.erl [hunk 4]
```diff
@@ -118,6 +118,8 @@ test_sse2_enlisted(_Config) ->
     rpc(?NODE2, sse, start, []),
   File =
```
Comment (jacobythwaites) [inline apps/sse/ct_test/sse_SUITE.erl]: I'd forgotten about this.

File: rel/files/release/sse.sh [hunk 5]
```diff
@@ -20,7 +20,7 @@
 # The SSE_LIB env var must point to the installed
 # sse library directory, such as /usr/local/lib/sse
 #
-ERL=${SSE_LIB}/bin/erl
+ERL="${SSE_LIB}/bin/erl -kernel logger_sasl_compatible true"
```
Comment (jacobythwaites) [inline rel/files/release/sse.sh]: I guess we should update our use of logging sometime. But not important.

**Pull request merged**